pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/MeasuresFunctors.hpp

```diff
@@ -16,27 +16,455 @@
 
     KOKKOS_INLINE_FUNCTION
-    void operator()(const std::size_t k) const {
-        const PrecisionT REAL = arr(k).real();
-        const PrecisionT IMAG = arr(k).imag();
-        probability(k) = REAL * REAL + IMAG * IMAG;
+    void init(PrecisionT dst[]) const {
+        for (unsigned i = 0; i < value_count; ++i)
+            dst[i] = 0.0;
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void join(PrecisionT dst[], const PrecisionT src[]) const {
+        for (unsigned i = 0; i < value_count; ++i)
+            dst[i] += src[i];
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()(const std::size_t i, const std::size_t j,
+                    PrecisionT dst[]) const {
+        const std::size_t index = all_indices(i) + all_offsets(j);
+        const PrecisionT rsv = arr(index).real();
+        const PrecisionT isv = arr(index).imag();
+        dst[i] += rsv * rsv + isv * isv;
+    }
+};
+
+/**
+ * @brief Compute probability distribution from StateVector.
+ *
+ * @tparam PrecisionT StateVector precision.
+ * @tparam DeviceType Kokkos execution space.
+ * @tparam num_wires Number of wires (0 is used for a dynamic number of wires).
+ * @param arr_ StateVector data.
+ * @param num_qubits_ Number of qubits.
+ * @param wires_ Wires for which the probability is computed.
+ */
+template <class PrecisionT, class DeviceType, std::size_t num_wires>
+class getProbsNQubitOpFunctor {
+  public:
+    // Required for functor:
+    using execution_space = DeviceType;
+    using value_type = PrecisionT[];
+    const unsigned value_count;
+
+    using UnmanagedSizeTHostView =
+        Kokkos::View<std::size_t *, Kokkos::HostSpace,
+                     Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+    using ComplexT = Kokkos::complex<PrecisionT>;
+    using KokkosComplexVector = Kokkos::View<ComplexT *>;
+    Kokkos::View<ComplexT *> arr;
+    const std::size_t n_wires;
+
+    Kokkos::View<std::size_t *> parity;
+    Kokkos::View<std::size_t *> rev_wires;
+
+    std::size_t rev_wire_0;
+    std::size_t rev_wire_1;
+    std::size_t rev_wire_2;
+    std::size_t rev_wire_3;
+    std::size_t rev_wire_4;
+    std::size_t rev_wire_5;
+    std::size_t rev_wire_6;
+    std::size_t rev_wire_7;
+    std::size_t rev_wire_8;
+    std::size_t parity_0;
+    std::size_t parity_1;
+    std::size_t parity_2;
+    std::size_t parity_3;
+    std::size_t parity_4;
+    std::size_t parity_5;
+    std::size_t parity_6;
+    std::size_t parity_7;
+    std::size_t parity_8;
+    std::size_t parity_9;
+
+    getProbsNQubitOpFunctor(const Kokkos::View<ComplexT *> &arr_,
+                            const std::size_t num_qubits_,
+                            const std::vector<std::size_t> &wires_)
+        : value_count{1U << wires_.size()}, arr{arr_}, n_wires{wires_.size()} {
+        PL_ABORT_IF(num_wires != 0 && num_wires != n_wires,
+                    "num_wires must be equal to n_wires.");
+        const std::size_t n_wires = wires_.size();
```

Comment:
Do we have to redeclare `n_wires` here? The initial declare is at Line 97.

Reply:
Good catch, forgot to remove it. Thx.